### PR TITLE
Janitor Lightreplacer replaces at a distance

### PR DESCRIPTION
**Pull request by etherware-novice**
# About The Pull Request
Edits the Light Replacer to work at a distance, using the same sfx and visuals as the Bluespace RPED 

## Why It's Good For The Game
Makes janitors job just a touch easier, as you dont need to waste time getting the crew to let you into restricted areas to replace lights as much. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
tweak: made the light replacer work through walls
/:cl:

### CODE REVIEW

**Comment by nednaZ** [comment]
This has the catgloves change in it as well.
Update your master branch.

**Comment by etherware-novice** [comment]
> This has the catgloves change in it as well.
> 
> Update your master branch.

how do i update the master branch??

**Comment by nednaZ** [comment]
> how do i update the master branch??
Go to your local branch in VSC and hit update.

**Comment by etherware-novice** [comment]
> > how do i update the master branch??
> > Go to your local branch in VSC and hit update.

please help i tried but it didnt say there were any changes to commit

**Comment by Serxule** [comment]
probably have this be an upgrade, bluespace light replacer, like the bluespace rped

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -115,6 +130,9 @@
 		var/found_lightbulbs = FALSE
 		var/replaced_something = TRUE
 
+
+
```
[the remark this answers is not in the document]
Comment (etherware-novice) [reply code/game/objects/items/devices/lightreplacer.dm:134]: it was stuff i had added then subsequently removed i dont know how to get rid of it its fine-

File: code/game/objects/items/devices/lightreplacer.dm [hunk 2]
```diff
@@ -65,12 +66,26 @@
 	// when we get this many shards, we get a free bulb.
 	var/shards_required = 4
 
+
+
+
+//monkestation edit  (using brped)
```
[the remark this answers is not in the document]
Comment (etherware-novice) [reply code/game/objects/items/devices/lightreplacer.dm:72]: i thought i was supposed to mark monkeststion edits thats what zanden said

File: code/game/objects/items/devices/lightreplacer.dm [hunk 3]
```diff
@@ -65,12 +65,21 @@
 	// when we get this many shards, we get a free bulb.
 	var/shards_required = 4
 
+//monkestation edit  (using brped)
+/obj/item/lightreplacer/afterattack(obj/item/itemtarget, mob/living/user)
+	if(istype(itemtarget, /obj/machinery/light))
+		user.Beam(itemtarget, icon_state = "rped_upgrade", time = 5)
+		playsound(src, 'sound/items/pshoom.ogg', 40, 1)
+		attack_self(itemtarget)
```
Comment (nednaZ) [inline code/game/objects/items/devices/lightreplacer.dm:73]: There is a dedicated proc for this. "ReplaceLight"

File: code/game/objects/items/devices/lightreplacer.dm [hunk 4]
```diff
@@ -65,12 +65,21 @@
 	// when we get this many shards, we get a free bulb.
 	var/shards_required = 4
 
+//monkestation edit  (using brped)
```
Comment (nednaZ) [inline code/game/objects/items/devices/lightreplacer.dm:68]: Mark both the beginning and end of monkestation edit blocks with comments.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 5]
```diff
@@ -65,12 +65,21 @@
...
+/obj/item/lightreplacer/afterattack(obj/item/itemtarget, mob/living/user)
```
Comment (nednaZ) [inline code/game/objects/items/devices/lightreplacer.dm:69]: You have this double defined.
Make it work in the already existing afterattack proc.

File: code/game/objects/items/devices/lightreplacer.dm [hunk 6]
```diff
@@ -65,12 +65,21 @@
...
+	if(istype(itemtarget, /obj/machinery/light))
+		user.Beam(itemtarget, icon_state = "rped_upgrade", time = 5)
```
Comment (nednaZ) [inline code/game/objects/items/devices/lightreplacer.dm:71]: Only play the beam and sound if at a distance, currently it does so even up close.

**Review by nednaZ — CHANGES_REQUESTED** [review]
It works, but doesn't tell the user that there is already a functioning bulb in the light unless up close.

**Pull request closed without merging**